lib/syntax.ml

```diff
@@ -278,6 +287,12 @@ and assign_types expr =
         | Ret ret, _ ->
           let nullability = common_nullability types in
           { t = ret; nullability; }, types (* ignoring arguments FIXME *)
+        | Comparison, _  ->
+          let args, ret = convert_args (Typ (depends Bool)) [Var 0; Var 0] in
+          let nullable = common_nullability args in
+          if param_strict && List.exists is_param params then 
```

Comment:
why List.exists is_param? I would expect this logic applies regardless if there are params or not

Reply:
Yes, I realized that this is useless checking, removed